repos/system_upgrade/el7toel8/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -127,20 +127,19 @@ def gather_target_repositories(target_rhsm_info):
                 }
             )
 
-    target_repoids = []
+    target_repoids = set()
```

Comment:
This is to avoid duplicate repoids being added to the list.

Reply:
just for info, usually is better to work with list and than create set. Here it is not so big deal, but in case you modfy set manytimes and set is going to be big,  you will reach performance issues.

Reply:
You are right. I tried that with more complex structures and direct use of set is more efficient than hack mentioned above.  Maybe it was wrong all the time, maybe it was correct when set was not built-in yet in python and you had to use the `sets` module (no longer exists in Py3, still can be found in Py2). Anyway direct use of the `set` is faster today.

Interesting is, that Python 2.7 is even faster than Python3.7 (in both cases, list and test as well) around 3.5s (tried several times). I expected that Python3 will be more efficient.